src/cobra/test/test_util/test_context.py

```diff
@@ -0,0 +1,42 @@
+"""Test functions of context.py."""
+
+import pytest
+
+from cobra.util import HistoryManager, get_context, resettable
+
+
+def test_history_manager():
+    """Test initialization and resetting of HistoryManager."""
+    # initialize HistoryManager
+    history_manager = HistoryManager()
+    # add non-functioning operation
+    history_manager(lambda: 1)
+    # reset operations
+    history_manager.reset()
+
+
+def test_get_context(model):
+    """Test if context retrieval is working."""
+    with model as model:
+        # reverse optimization direcion
+        model.objective_direction = "min"
+        context = get_context(model)
+        # should have history
+        assert context._history[-1].args[-1] == "max"
```

Comment:
This seems brittle to me. We should not access private class members. You could rather assert that before and after the context, the `objective_direction` is `"max"`.

Reply:
I understand that accessing private class members, is not a good practice. But, if we just check the values before and after the context, we are not directly checking the functionality of the `get_context()` function but rather indirectly verifying whether the thing works as a whole. Will that indirect test be enough?

Reply:
The problem I see is that the test relies on a lot of class internal specifics which it shouldn't. You could implement a `__len__` method that returns the number of commands on the history stack and test that, for example?

Reply:
That is a better way. I'll implement that.

Reply:
Done.